From ba99c63cd20f18d6205d7968ed7b0032b5dfea42 Mon Sep 17 00:00:00 2001
From: Gaku Inami <gaku.inami.xw@bp.renesas.com>
Date: Tue, 31 Jan 2017 14:52:30 +0900
Subject: [PATCH 1727/2066] phy: rcar-gen3-usb2: Add the interrupt disabled in
 suspend

Since the interrupt of USB2 phy can't be handled in suspend sequence,
it disables the interrupt in suspend callback.

Signed-off-by: Gaku Inami <gaku.inami.xw@bp.renesas.com>
Signed-off-by: Kazuya Mizuguchi <kazuya.mizuguchi.ks@renesas.com>
[zou:Original patch taken from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git
 v4.9/rcar-3.5.1]
Signed-off-by: zou cao <cao.zou@windriver.com>
---
 drivers/phy/phy-rcar-gen3-usb2.c |   13 ++++++++++---
 1 files changed, 10 insertions(+), 3 deletions(-)

diff --git a/drivers/phy/phy-rcar-gen3-usb2.c b/drivers/phy/phy-rcar-gen3-usb2.c
index 865b4f2..d493840 100644
--- a/drivers/phy/phy-rcar-gen3-usb2.c
+++ b/drivers/phy/phy-rcar-gen3-usb2.c
@@ -1,7 +1,7 @@
 /*
  * Renesas R-Car Gen3 for USB2.0 PHY driver
  *
- * Copyright (C) 2015 Renesas Electronics Corporation
+ * Copyright (C) 2015-2017 Renesas Electronics Corporation
  *
  * This is based on the phy-rcar-gen2 driver:
  * Copyright (C) 2014 Renesas Solutions Corp.
@@ -481,8 +481,15 @@ static int rcar_gen3_phy_usb2_remove(struct platform_device *pdev)
 #ifdef CONFIG_PM_SLEEP
 static int rcar_gen3_phy_suspend(struct device *dev)
 {
-	/* Empty function for now */
-	return 0;
+	int ret = 0;
+
+	if (to_phy(dev))
+		ret = rcar_gen3_phy_usb2_exit(to_phy(dev));
+	else {
+		pr_warn("%s: No phy dev\n", __func__);
+		ret = -ENODEV;
+	}
+	return ret;
 }
 
 static int rcar_gen3_phy_resume(struct device *dev)
-- 
1.7.5.4

